filemanager/__init__.py

```diff
@@ -16,290 +13,437 @@
+                filename = (
+                    self.basepath
+                    + self.current_path
+                    + os.path.basename(path)
+                )
+                if os.path.exists(filename):
+                    messages.append(
+                        'ERROR: A file/folder with this name already exists in'
+                        + ' the destination folder.'
+                    )
+                else:
+                    if action == 'move':
+                        method = shutil.move
+                    else:
+                        if file_or_dir == 'dir':
+                            method = shutil.copytree
+                        else:
+                            method = shutil.copy
+                    try:
+                        method(self.basepath+path, filename)
+                    except:
+                        messages.append(
+                            'File/folder couldn\'t be moved/copied.'
+                        )
+        return messages
 
-  def directory_structure(self):
-    self.idee = 0
-    dir_structure = {'':{'id':self.next_id(),'open':'yes','dirs':{},'files':[]}}
-    os.chdir(self.basepath)
-    for directory,directories,files in os.walk('.'):
-      directory_list = directory[1:].split('/')
-      current_dir = None
-      nextdirs = dir_structure
-      for d in directory_list:
-        current_dir = nextdirs[d]
-        nextdirs = current_dir['dirs']
-      if directory[1:]+'/' == self.current_path:
-        self.current_id = current_dir['id']
-      current_dir['dirs'].update(dict(map(lambda d:(d,{'id':self.next_id(),'open':'no','dirs':{},'files':[]}),directories)))
-      current_dir['files'] = files
-    return dir_structure
+    def directory_structure(self):
+        self.idee = 0
+        dir_structure = {
+            '': {
+                'id': self.next_id(),
+                'open': 'yes',
+                'dirs': {},
+                'files': [],
+            },
+        }
+        os.chdir(self.basepath)
+        for directory, directories, files in os.walk('.'):
+            directory_list = directory[1:].split('/')
+            current_dir = None
+            nextdirs = dir_structure
+            for d in directory_list:
+                current_dir = nextdirs[d]
+                nextdirs = current_dir['dirs']
+            if directory[1:]+'/' == self.current_path:
+                self.current_id = current_dir['id']
+            current_dir['dirs'].update(
+                dict(
+                    map(
+                        lambda d: (
+                            d,
+                            {
+                                'id': self.next_id(),
+                                'open': 'no',
+                                'dirs': {},
+                                'files': [],
+                            }
+                        ),
+                        directories,
+                    )
+                )
+            )
```

Comment:
@ErwinJunge. The line breaks here. I personally feel this level of breaking is not required. What do you think?

Reply:
I actually prefer this codestyle. I think it makes it clearer where the dict, map and lambda end. However, codestyle is 90% personal preference and this is your package so your preference wins. What style would you prefer?

```
                dict(map(lambda d: (d, {
                    'id': self.next_id(),
                    'open': 'no',
                    'dirs': {},
                    'files': [],
                }), directories)))
```
Something like this?

Reply:
You can leave it for now. Looks fine. Will change it sometime later if needed. :smiley: